process_subways.py

```diff
@@ -71,114 +72,152 @@ def slugify(name):
+
+
 def calculate_centers(elements):
     """Adds 'center' key to each way/relation in elements,
     except for empty ways or relations.
     Relies on nodes-ways-relations order in the elements list.
     """
-    nodes = {}  # id(int) => (lat, lon)
-    ways = {}  # id(int) => (lat, lon)
-    relations = {}  # id(int) => (lat, lon)
-    empty_relations = set()  # ids(int) of relations without members
-    # or containing only empty relations
-
-    def calculate_way_center(el):
-        # If element has been queried via overpass-api with 'out center;'
-        # clause then ways already have 'center' attribute
-        if "center" in el:
-            ways[el["id"]] = (el["center"]["lat"], el["center"]["lon"])
-            return
-        center = [0, 0]
-        count = 0
-        way_nodes = el["nodes"]
-        way_nodes_len = len(el["nodes"])
-        for i, nd in enumerate(way_nodes):
-            if nd not in nodes:
-                continue
-            # Don't count the first node of a closed way twice
-            if (
-                i == way_nodes_len - 1
-                and way_nodes_len > 1
-                and way_nodes[0] == way_nodes[-1]
-            ):
-                break
-            center[0] += nodes[nd][0]
-            center[1] += nodes[nd][1]
-            count += 1
-        if count > 0:
-            el["center"] = {"lat": center[0] / count, "lon": center[1] / count}
-            ways[el["id"]] = (el["center"]["lat"], el["center"]["lon"])
-
-    def calculate_relation_center(el):
-        # If element has been queried via overpass-api with 'out center;'
-        # clause then some relations already have 'center' attribute
-        if "center" in el:
-            relations[el["id"]] = (el["center"]["lat"], el["center"]["lon"])
-            return True
-        center = [0, 0]
-        count = 0
-        for m in el.get("members", []):
-            if m["type"] == "relation" and m["ref"] not in relations:
-                if m["ref"] in empty_relations:
-                    # Ignore empty child relations
-                    continue
-                else:
-                    # Center of child relation is not known yet
-                    return False
-            member_container = (
-                nodes
-                if m["type"] == "node"
-                else ways
-                if m["type"] == "way"
-                else relations
-            )
-            if m["ref"] in member_container:
-                center[0] += member_container[m["ref"]][0]
-                center[1] += member_container[m["ref"]][1]
-                count += 1
-        if count == 0:
-            empty_relations.add(el["id"])
-        else:
-            el["center"] = {"lat": center[0] / count, "lon": center[1] / count}
-            relations[el["id"]] = (el["center"]["lat"], el["center"]["lon"])
-        return True
+    nodes: Dict[int, Tuple[float, float]] = {}  # id => (lat, lon)
```

Comment:
You can create an alias for `Tuple[float, float]` like `Pin = Tuple[float, float]`, and use it instead:
```python
Pin = Tuple[float, float]
nodes: Dict[int, Pin] = {} # id => (lat, lon)
```

Reply:
\+